src/plugins/discover/server/locator/title_from_locator.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { SavedObject } from '@kbn/core/server';
+import { LocatorServicesDeps } from '.';
+import { DiscoverAppLocatorParams } from '../../common';
+
+/**
+ * @internal
+ */
+export const titleFromLocatorFactory = (services: LocatorServicesDeps) => {
+  /**
+   * Allows consumers to derive a title of a search in Disocver from DiscoverAppLocatorParams
+   *
+   * @public
+   */
+  const titleFromLocator = async (params: DiscoverAppLocatorParams): Promise<string> => {
+    const { savedObjects } = services;
+    const searchObject: SavedObject<{ title?: string }> = await savedObjects.get(
+      'search',
+      params.savedSearchId! // TODO: allow default to DavaView name
```

Comment:
This is related to https://github.com/elastic/kibana/pull/150631/files#r1103440151

The `TODO` explains that the code won't work if the DiscoverAppLocatorParams does not have a reference to a saved search. My thinking is this can stay as a TODO for now, since the current use case for this API is to power exports using DiscoverAppLocatorParams that do contain a saved search ID.